ambry-router/src/main/java/com.github.ambry.router/AdaptiveOperationTracker.java

```diff
@@ -83,7 +81,7 @@
       localDcResourceToHistogram = getResourceToLatencyMap(routerOperation, true);
       crossDcResourceToHistogram = getResourceToLatencyMap(routerOperation, false);
     }
-    if (parallelism > routerConfig.routerOperationTrackerMaxInflightRequests) {
+    if (Math.max(diskParallelism, cloudParallelism) > routerConfig.routerOperationTrackerMaxInflightRequests) {
```

Comment:
I guess the reason we introduced `routerOperationTrackerMaxInflightRequests` is to throttle total # of request in regular data centers. So, although current number of disk requests equals `MaxInflightRequests`, router is still able to route request to cloud which is not restricted by `MaxInflightRequests` for regular dc(s). 
I agree with one point that we'd better break it into different checks and probably use separate `MaxInflightRequests` for cloud dc and regular dc.

Reply:
Yep, to expand on this, the adaptive operation tracker has the ability to go beyond the parallelism setting if requests are past due. This limit makes sure that it does not snowball and keep sending out parallel requests.

Reply:
I added a second if for the constructor validations but kept the logic as is. Since max in flight requests is used when the number of requests made needs to go beyond the initial parallelism for performance reasons but isn't related to the quorum, I kept the config as is so that all in flight requests count towards the same limit.